control/xferfcn.py

```diff
@@ -1349,3 +1317,42 @@ def tfdata(sys):
     tf = _convertToTransferFunction(sys)
 
     return (tf.num, tf.den)
+
+def _cleanPart(data):
+    '''
+    Return a valid, cleaned up numerator or denominator 
+    for the TransferFunction class.
+    
+    Parameters:
+    data: numerator or denominator of a transfer function.
+    
+    Returns:
+    data: correctly formatted transfer function part.
+    ;
+    '''
+    valid_types = (int, int8, int16, int32, int64,
+                   float, float16, float32, float64, float128)
+    valid_collection = (list, tuple, ndarray)
+
+    if (isinstance(data, valid_types) or
```

Comment:
I was going to suggest using `np.isreal` instead of `isinstance(..,valid_types)`, but then I found `np.isreal(file('/dev/null'))` returns True (oops!).